compiler/noirc_frontend/src/hir/comptime/interpreter.rs

```diff
@@ -1595,15 +1595,29 @@ fn evaluate_prefix_with_value(rhs: Value, operator: UnaryOp, location: Location)
     match operator {
         UnaryOp::Minus => match rhs {
             Value::Field(value) => Ok(Value::Field(-value)),
-            Value::I8(value) => Ok(Value::I8(-value)),
-            Value::I16(value) => Ok(Value::I16(-value)),
-            Value::I32(value) => Ok(Value::I32(-value)),
-            Value::I64(value) => Ok(Value::I64(-value)),
-            Value::U8(value) => Ok(Value::U8(0 - value)),
-            Value::U16(value) => Ok(Value::U16(0 - value)),
-            Value::U32(value) => Ok(Value::U32(0 - value)),
-            Value::U64(value) => Ok(Value::U64(0 - value)),
-            Value::U128(value) => Ok(Value::U128(0 - value)),
+            Value::I8(value) => value
+                .checked_neg()
+                .map(Value::I8)
+                .ok_or_else(|| InterpreterError::NegateWithOverflow { location }),
+            Value::I16(value) => value
+                .checked_neg()
+                .map(Value::I16)
+                .ok_or_else(|| InterpreterError::NegateWithOverflow { location }),
+            Value::I32(value) => value
+                .checked_neg()
+                .map(Value::I32)
+                .ok_or_else(|| InterpreterError::NegateWithOverflow { location }),
+            Value::I64(value) => value
+                .checked_neg()
+                .map(Value::I64)
+                .ok_or_else(|| InterpreterError::NegateWithOverflow { location }),
+            Value::U8(_) => Err(InterpreterError::CannotApplyMinusToType { location, typ: "u8" }),
```

Comment:
I think we're missing `Value::U1` here (and below), can you add it for consistency?

Reply:
I added it locally, then noticed it's also missing when evaluating `for`, so I went to try it and I get this error message:

```
The type `u1` is currently unsupported in comptime expressions
```

Do you know why that is?

Reply:
> Do you know why that is?

I think it is just that it was added afterward and wasn't supported initially but not all match cases were changed to account for this.

Reply:
Done! I added support for u1 and fixed a few things.